lib/src/barback/dartdevc/dartdevc_module_transformer.dart

```diff
@@ -0,0 +1,120 @@
+        allAssetIds..addAll(module.assetIds)..addAll(linkedSummaryIds);
+      }
+      // Create a single temp environment for all the modules in this package.
+      tempEnv = await TempEnvironment.create(allAssetIds, transform.readInput);
+      var outputDir = topLevelDir(transform.primaryInput.id.path);
+      await Future.wait(modules.map((m) => _createDartdevcModule(
+          m, outputDir, tempEnv, summariesForModule[m.id], transform)));
+    } finally {
+      tempEnv?.delete();
+    }
+  }
+}
+
+/// Compiles [module] using the `dartdevc` binary from the SDK to a relative
+/// path under the package that looks like `$outputDir/${module.id.name}.js`.
+Future _createDartdevcModule(
+    Module module,
+    String outputDir,
+    TempEnvironment tempEnv,
+    Set<AssetId> linkedSummaryIds,
+    Transform transform) async {
+  var logger = transform.logger;
+  var jsOutputId = new AssetId(
+      module.id.package, p.url.join(outputDir, '${module.id.name}.js'));
+  var jsOutputFile = tempEnv.fileFor(jsOutputId);
+  var sdk_summary = p.url.join(sdkDir.path, 'lib/_internal/ddc_sdk.sum');
+  var request = new WorkRequest();
+  request.arguments.addAll([
+    '--dart-sdk-summary=$sdk_summary',
+    // TODO(jakemac53): Remove when no longer needed.
+    '--unsafe-angular2-whitelist',
+    '--modules=amd',
+    '--dart-sdk=${sdkDir.path}',
+    '--module-root=${tempEnv.tempDir.path}',
+    '--library-root=${p.dirname(jsOutputFile.path)}',
+    '--summary-extension=${linkedSummaryExtension.substring(1)}',
+    '-o',
+    jsOutputFile.path,
+  ]);
+  // Add all the linked summaries as summary inputs.
+  for (var id in linkedSummaryIds) {
+    request.arguments.addAll(['-s', tempEnv.fileFor(id).path]);
+  }
+  // Add url mappings for all the package: files to tell ddc where to find them.
+  for (var id in module.assetIds) {
+    var uri = canonicalUriFor(id);
+    if (uri.startsWith('package:')) {
+      request.arguments.add('--url-mapping=$uri,${tempEnv.fileFor(id).path}');
+    }
+  }
+  // And finally add all the urls to compile, using the package: path for files
+  // under lib and the full absolute path for other files.
+  request.arguments.addAll(module.assetIds.map((id) {
+    var uri = canonicalUriFor(id);
+    if (uri.startsWith('package:')) {
+      return uri;
+    }
+    return tempEnv.fileFor(id).path;
+  }));
+
+  var response = await dartdevcDriver.doWork(request);
+
+  // TODO(jakemac53): Fix the ddc worker mode so it always sends back a bad
+  // status code if something failed. Today we just make sure there is an output
+  // js file to verify it was successful.
+  if (response.exitCode != EXIT_CODE_OK || !jsOutputFile.existsSync()) {
+    // We only log warnings for ddc modules because technically they don't all
+    // need to compile successfully, only the ones imported by an entrypoint do.
+    logger.warning('Error compiling dartdevc module: ${module.id}.\n'
+        '${response.output}');
```

Comment:
This seems... dubious to me. I see the value in it, but I'm not sold.

By that same argument, we could warn if there are errors in methods, as long as you don't call them. But, in a statically typed language, I think users expect "zero errors" to mean "all the code correctly compiled".

Do you know for certain that we really need to be loose here? Is there a way we can let users control it?